qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.kt

```diff
@@ -107,22 +108,63 @@ class MainActivity : CrashReportingActivity(),
     private var lastBackPressTime = 0L
     private val backPressConfirmInterval = 2000L
 
+    private val profileIconSize: Int
+        get() = ctx.resources.getDimension(
+                R.dimen.material_drawer_item_profile_icon_width).toInt()
 
-    private val accountBroadCastReceiver = object : BroadcastReceiver() {
-        override fun onReceive(context: Context, intent: Intent) {
-            val statusCode = intent.getIntExtra(QblBroadcastConstants.STATUS_CODE_PARAM, -1)
-            when (statusCode) {
-                AccountStatusCodes.LOGOUT -> navigator.selectCreateAccountActivity()
-            }
-        }
-    }
-
-    private val chatBroadCastReceiver = object : BroadcastReceiver() {
-        override fun onReceive(context: Context?, intent: Intent?) {
-            updateNewMessageBadge()
-        }
 
-    }
+    override val intentListeners = listOf(
+            intentListener(QblBroadcastConstants.Account.ACCOUNT_CHANGED, {
+                val statusCode = it.getIntExtra(QblBroadcastConstants.STATUS_CODE_PARAM, -1)
+                when (statusCode) {
+                    AccountStatusCodes.LOGOUT -> finish()
+                }
+            }),
+            intentListener(QblBroadcastConstants.Chat.MESSAGE_STATE_CHANGED, {
+                updateNewMessageBadge()
+            }),
+            intentListener(QblBroadcastConstants.Identities.IDENTITY_CHANGED) {
```

Comment:
Pretty sure that the IDENTITY_CHANGED and IDENTITY_REMOVED listeners are untested.
By extracting these anonymous functions to separate classes, you could probably test them more easily, and let the reader find the tests. Additionally, it will be easier to flesh out the main activity and move methods like `updateNewMessageBadge` to more cohesive classes if these listeners have their dependencies managed (explicit and injected).